libraries/AP_HAL_PX4/CAN.h

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (C) 2015-2016  Intel Corporation. All rights reserved.
+ *
+ * This file is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This file is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#pragma once
+
+#include "AP_HAL_PX4.h"
+#include <systemlib/perf_counter.h>
+#include <AP_HAL/CAN.h>
+
+class PX4::PX4CAN : public AP_HAL::CAN {
+public:
+    PX4CAN() { }
+    ~PX4CAN() { }
+    int16_t send(const uavcan::CanFrame& frame, uavcan::MonotonicTime tx_deadline, uavcan::CanIOFlags flags);
+    int16_t receive(uavcan::CanFrame& out_frame, uavcan::MonotonicTime& out_ts_monotonic, uavcan::UtcTime& out_ts_utc,
+            uavcan::CanIOFlags& out_flags);
+    int16_t configureFilters(const uavcan::CanFilterConfig* filter_configs, uint16_t num_configs);
+    uint16_t getNumFilters() const;
+    uint64_t getErrorCount() const;
+    bool begin(uint32_t bitrate);
+    bool begin(uint32_t bitrate, float samplepoint);
+    void end();
+    void reset();
+    bool is_initialized();
+    int32_t tx_pending();
+    uint32_t available();
+
+private:
+    uint32_t _baudrate;
+    volatile bool _initialised;
+    volatile bool _in_timer;
+    perf_counter_t  _perf_can;
```

Comment:
Fun! Uninitialized fields! I take it this is just a prototype, and you will add field initializers later.